### DISCUSSION TITLE
Relying on local knowledge

### DISCUSSION THREAD

**Discussion by tianchiguaixia** [explosion/spacy-llm#354] [Q&A]
When doing NER. How to share local knowledge with the model to improve effectiveness

**Comment by rmitsch** [comment]
Converting this to a discussion.

**Comment by rmitsch** [comment]
Hi @tianchiguaixia! Can you elaborate on which kind of local knowledge you would like to share and how you think this would improve NER accuracy?

**Comment by tianchiguaixia** [comment]


**Reply by rmitsch** [reply]
Please state in text what this local knowledge looks like and how you think it can improve NER performance.

**Comment by tianchiguaixia** [comment]
It is a JSON structure with corresponding key value values. But there are no start and end.

**Reply by rmitsch** [reply]
I'm not sure how this JSON structure relates to the entities you want to recognize, but you could include some data as [examples](https://spacy.io/usage/large-language-models#few-shot-prompts) in the prompt.

Beyond that there is no possibility to directly incorporate this information into the LLM task. I recommend looking at the other components in [spaCy](https://spacy.io/usage/spacy-101) to see how this can be achieve - you could  use e. g. a rule-based matcher or train your own NER model.

**Comment by tianchiguaixia** [comment]
Your suggestion is very useful to me
